### PR TITLE
Fixes several resource leaks listed on coverity

### PR DESCRIPTION
**Pull request by Crazy-P**
Fixes resource leaks in:
- gdscript (CID#269851)
- gdscript compiler  (CID#269423)
- regex  (CID#269571 &  CID#269860)
- android/export  (CID#269470)
- gles3/rasterizer scene (CID#269848)
- gles3/rasterizer storage (CID#269660)

### CODE REVIEW

File: modules/gdscript/gdscript_compiler.cpp [hunk 1]
```diff
@@ -1322,6 +1322,8 @@ Error GDScriptCompiler::_parse_block(CodeGen &codegen, const GDScriptParser::Blo
 
 						int ret = _parse_expression(codegen, op, p_stack_level);
 						if (ret < 0) {
+							memdelete(op);
```
Comment (vnen) [inline modules/gdscript/gdscript_compiler.cpp]: It should delete the `id` variable as well, doesn't it?

File: modules/gdscript/gdscript.cpp [hunk 2]
```diff
@@ -719,22 +719,32 @@ Error GDScript::load_byte_code(const String &p_path) {
...
 		key.resize(32);
 		for (int i = 0; i < key.size(); i++) {
 			key.write[i] = script_encryption_key[i];
 		}
+
 		Error err = fae->open_and_parse(fa, key, FileAccessEncrypted::MODE_READ);
-		ERR_FAIL_COND_V(err, err);
+
+		if (err) {
+			memdelete(fae);
```
Comment (vnen) [inline modules/gdscript/gdscript.cpp]: I think it should delete `fa` as well, with or without error.

**Comment by akien-mga** [comment]
Thanks a lot!

File: modules/gdscript/gdscript.cpp [hunk 3]
```diff
@@ -126,7 +126,10 @@ GDScriptInstance *GDScript::_create_instance(const Variant **p_args, int p_argco
 		GDScriptLanguage::singleton->lock->unlock();
 #endif
 
-		ERR_FAIL_COND_V(r_error.error != Variant::CallError::CALL_OK, NULL); //error constructing
+		if (r_error.error != Variant::CallError::CALL_OK) {
+			memdelete(instance);
```
Comment (vnen) [inline modules/gdscript/gdscript.cpp]: This one causes a crash, the instance is actually deleted earlier (when it does `instance->owner->set_script_instance(NULL);`). I'll remove it together with other GDScript fixes.

**Pull request merged**